### DISCUSSION TITLE
Network issue during installation with RAUC streaming

### DISCUSSION THREAD

**Discussion by hatmaalla** [rauc/rauc#1023] [Q&A]
Hello,
I am a beginner with RAUC 1.8 and I am using the streaming capabilities in my project.
In order to handle some alternative cases, I am asking if RAUC can resume the installation process after the network connection fails or the process is totally interrupted and I have to start the process from the beginning?

**Comment by jluebbe** [comment]
RAUC will retry individual failed requests a few times, but then the installation will be aborted.

At the moment, we don't update the hash index of the target slot for each written block in `copy_block_hash_index_image_to_dev()`, so it will not detect that some block have been written already on subsequent attempts. But that could be improved.

**Reply by hatmaalla** [reply]
Thank you for your quick response,
As I can understand, if the connection is resumed, RAUC will rewrite some blocks although these blocks are already flashed, but not all blocks.
Another question, is there an explicit message sent via the dbus interface to indicate the status of the network (exemple: "http connection failed", "retry connection in progress", etc...)?
